cmd/install/assets/hypershift_operator.go

```diff
@@ -863,6 +864,36 @@ func (o ExternalDNSClusterRoleBinding) Build() *rbacv1.ClusterRoleBinding {
 	return binding
 }
 
+type ExternalDNSPodMonitor struct {
+	Namespace *corev1.Namespace
+}
+
+func (o ExternalDNSPodMonitor) Build() *prometheusoperatorv1.PodMonitor {
+	return &prometheusoperatorv1.PodMonitor{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PodMonitor",
+			APIVersion: prometheusoperatorv1.SchemeGroupVersion.String(),
```

Comment:
To note https://github.com/openshift/hypershift/pull/1889 should flip this to either rhobs or coreos apigroups depending on the `RHOBSMonitoring` option. This works the same way as the [HyperShift operator servicemonitor](https://github.com/openshift/hypershift/blob/591b9e18ef994d1d967df043aaf170c5dc0045bb/cmd/install/assets/hypershift_operator.go#L1379-L1408).